Author: bgstack15
Purpose: patch mineclonia 0.112.2 to work with mod molten_sailor_mcl
Application: mineclonia at https://codeberg.org/mineclonia/mineclonia
Version: 0.112.2
Last-modified: 2025-03-13
--- mineclonia/mods/ENTITIES/mcl_burning/api.lua	   2025-03-07 10:43:26.377727853 -0500
+++ mineclonia.new/mods/ENTITIES/mcl_burning/api.lua	2025-03-13 18:27:09.976537113 -0400
@@ -1,4 +1,5 @@
 local enable_damage = minetest.settings:get_bool("enable_damage")
+local has_molten_sailor = minetest.get_modpath("molten_sailor_mcl") ~= nil
 
 local collisionbox_cache = {}
 
@@ -103,8 +104,14 @@
 		return
 	end
 
-	if obj:is_player() and not enable_damage then
-		return
+	if obj:is_player() then
+      if not enable_damage then
+         return
+      end
+		-- lava suit protects from setting on fire
+		if has_molten_sailor and molten_sailor_mcl.has_full_lava_suit(obj) then
+			return
+		end
 	else
 		local max_fire_prot_lvl = 0
 		local inv = mcl_util.get_inventory(obj)
--- mineclonia/mods/CORE/mcl_damage/init.lua    	2025-03-07 10:43:26.369727831 -0500
+++ mineclonia.new/mods/CORE/mcl_damage/init.lua	2025-03-13 18:25:31.980053297 -0400
@@ -1,3 +1,5 @@
+local has_molten_sailor = minetest.get_modpath("molten_sailor_mcl") ~= nil
+
 mcl_damage = {
 	modifiers = {},
 	damage_callbacks = {},
@@ -264,7 +266,12 @@
 		local nodedef = minetest.registered_nodes[mt_reason.node]
 
 		if nodedef.damage_per_second then
-		  hp_change = -nodedef.damage_per_second
+			-- lava suit protects from taking direct damage from lava
+			if has_molten_sailor and nodedef.groups["lava"] ~= nil and molten_sailor_mcl.has_full_lava_suit(player) then
+				hp_change = 0
+			else
+				hp_change = -nodedef.damage_per_second
+			end
 		end
 	end
 
